### DISCUSSION TITLE
Galera, queria uma opinião sobre como chamar uma função em um componente filho no React?

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1856] [Perguntas e Respostas]
Fala galera, blz?

Estava fazendo uma popup aqui, e cheguei a um ponto em que uma ação do componente pai da popup deveria fechá-la, então usei ref para pegar a referencia da popup e fiz: ref.current.handleVisibility(false)... 

O que vcs acham de chamar métodos de componentes filhos pelos pais? É algo errado de se fazer? Como vcs fariam?
(Pensando q prezamos por uma programação descritiva me vez de uma imperativa com react...)

Muito obrigado!

**Comment by ninetails** [comment]
Se for uma ação que realmente necessita de uma função, seria esse o caminho.

Porém geralmente não tem ação a ser feito e seria somente um toggle como no caso do popup, vc pode ao invés de chamar uma função mudar apenas uma prop que o componente pai passa e que force o fechamento dela, controlando sua atualização dentro do próprio componente usando `useEffect` deixando essa prop no array do segundo argumento se for um componente funcional ou usando o [componentDidUpdate](https://reactjs.org/docs/react-component.html#componentdidupdate) se for um componente criado a partir de classe.
